src/main/java/roomescape/controller/AdminController.java

```diff
@@ -0,0 +1,58 @@
+package roomescape.controller;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import roomescape.domain.Reservation;
+import roomescape.domain.ReservationDto;
+import roomescape.service.AdminService;
+
+@Controller
+public class AdminController {
+
+    private final AdminService adminService;
+    private final AtomicLong index = new AtomicLong(1);
+
+    public AdminController() {
+        this.adminService = new AdminService();
+    }
+
+    @GetMapping("admin")
+    public String welcome() {
+        return "admin/index";
+    }
+
+    @GetMapping("admin/reservation")
+    public String reservation(Model model) {
+        model.addAttribute("reservations", adminService.getAllReservations());
+        return "admin/reservation-legacy";
+    }
+
+    @GetMapping("reservations")
+    @ResponseBody
+    public List<Reservation> reservations() {
+        return adminService.getAllReservations();
+    }
+
+    @PostMapping("reservations")
+    @ResponseBody
+    public Reservation reserve(@RequestBody ReservationDto reservationDto) {
+        Reservation reservation = new Reservation(index.getAndIncrement(), reservationDto);
+        adminService.addReservation(reservation);
+        return adminService.findReservation(reservation.getId());
+    }
+
+    @DeleteMapping("reservations/{id}")
+    public ResponseEntity<Void> cancel(@PathVariable("id") Long id){
```

Comment:
```suggestion
    public ResponseEntity<Void> cancel(@PathVariable long id) {
```

`@PathVariable`은 parameter이름과 같다면 굳이 name을 넣어주지 않아도 됩니다.

Long / long의 차이는 무엇인가요? 여기서는 무엇을 사용하는게 좋을까요?